x/trigger/types/genesis_test.go

```diff
@@ -0,0 +1,276 @@
+package types
+
+import (
+	"testing"
+	time "time"
+
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewGenesisState(t *testing.T) {
+	request := NewCreateTriggerRequest("addr", &BlockHeightEvent{}, []types.Msg{&MsgDestroyTriggerRequest{}})
+	trigger := NewTrigger(1, "owner", request.Event, request.Actions)
+	state := NewGenesisState(1, 2, []Trigger{trigger}, []GasLimit{{TriggerId: 1, Amount: 1}, {TriggerId: 2, Amount: 2}}, []QueuedTrigger{{BlockHeight: 1, Time: time.Time{}, Trigger: trigger}})
+
+	assert.Equal(t, uint64(1), state.TriggerId)
+	assert.Equal(t, uint64(2), state.QueueStart)
+	assert.Equal(t, []Trigger{trigger}, state.Triggers)
+	assert.Equal(t, []GasLimit{{TriggerId: 1, Amount: 1}, {TriggerId: 2, Amount: 2}}, state.GasLimits)
+	assert.Equal(t, []QueuedTrigger{{BlockHeight: 1, Time: time.Time{}, Trigger: trigger}}, state.QueuedTriggers)
+}
+
+func TestDefaultGenesis(t *testing.T) {
+	state := DefaultGenesis()
+
+	assert.Equal(t, uint64(1), state.TriggerId)
+	assert.Equal(t, uint64(1), state.QueueStart)
+	assert.Equal(t, []Trigger{}, state.Triggers)
+	assert.Equal(t, []GasLimit{}, state.GasLimits)
+	assert.Equal(t, []QueuedTrigger{}, state.QueuedTriggers)
+}
+
+func TestGenesisStateValidate(t *testing.T) {
+	request := NewCreateTriggerRequest("addr", &BlockHeightEvent{}, []types.Msg{&MsgDestroyTriggerRequest{Id: 1, Authority: "cosmos1w6t0l7z0yerj49ehnqwqaayxqpe3u7e23edgma"}})
+	badRequest := NewCreateTriggerRequest("addr", &TransactionEvent{Name: "", Attributes: []Attribute{}}, []types.Msg{&MsgDestroyTriggerRequest{Id: 1, Authority: ""}})
+	trigger := NewTrigger(1, "owner", request.Event, request.Actions)
+	trigger2 := NewTrigger(2, "owner", request.Event, request.Actions)
+
+	tests := []struct {
+		name   string
+		state  *GenesisState
+		modify func(*GenesisState)
+		err    string
+	}{
+		{
+			name:   "valid - genesis state validate",
+			state:  DefaultGenesis(),
+			modify: nil,
+			err:    "",
+		},
+		{
+			name: "invalid - trigger id cannot be zero",
+			state: &GenesisState{
+				TriggerId:      0,
+				QueueStart:     1,
+				GasLimits:      []GasLimit{},
+				Triggers:       []Trigger{},
+				QueuedTriggers: []QueuedTrigger{},
+			},
+			modify: nil,
+			err:    "invalid trigger id",
+		},
```

Comment:
There should be another test case in here checking behavior of `nil` for the slices.

I'm pretty sure that, in this case, `nil` and an empty slice should be treated the same. I.e. we don't need to return an error if one of the slices is nil instead of empty. That would depend on how they're used elsewhere though. I haven't seen anything that assumes the slice isn't nil, but don't know for certain that it's okay.

Something like this:
```golang
		{
			name: "valid - nil slices",
			state: &GenesisState{
				TriggerId:      1,
				QueueStart:     1,
				GasLimits:      nil,
				Triggers:       nil,
				QueuedTriggers: nil,
			},
			err: "",
		},
```

Reply:
Thanks, forgot about this case.